clang/test/SemaCXX/GH97087.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -std=c++20 -fsyntax-only %s
+// expected-no-diagnostics 
+struct X {};
+namespace NS {
+    bool operator==(X, X);
+}
+using namespace NS;
+struct Y {
+    X x;
+    friend bool operator==(Y, Y);
+};
+bool operator==(Y, Y) = default;
```

Comment:
Can you also add the following test: https://godbolt.org/z/hE3h4b4oM

Reply:
I think both tests should be put into `clang/test/SemaCXX/cxx20-default-compare.cpp` instead of creating a new file.